include/fmt/base.h

```diff
@@ -1902,7 +1902,9 @@ template <typename Context> class basic_format_args {
     if (id >= detail::max_packed_args) return arg;
     arg.type_ = type(id);
```

Comment:
Is this line that calls `type(id)` also a problem if `id` is out of bounds? What's the expected behavior? In the string-based wrapper below on line 1914 it returns an empty `format_arg()` up front. Should the check for `id >= 0` happen first? And then be removed from line 1914?

Not particularly familiar with this code, just wondering what the right bounds strategy is, since the static analysis tools are triggering for it.

Reply:
Yeah, type and value should either be both set or not. I commented above suggesting a better fix.